src/core/net/srp_client.cpp

```diff
@@ -524,6 +624,36 @@ Error Client::SetHostAddresses(const Ip6::Address *aAddresses, uint8_t aNumAddre
     return error;
 }
 
+void Client::HandleUnicastAddressEvent(Ip6::Netif::AddressEvent aEvent, const Ip6::Netif::UnicastAddress &aAddress)
+{
+    // This callback from `Netif` signals an impending addition or
+    // removal of a unicast address, occurring before `Notifier`
+    // events. If `AutoAddress` is enabled, we check whether the
+    // address origin is SLAAC (e.g., an OMR address) and request a
+    // longer `TxJitter`. This helps randomize the next SRP
+    // update transmission time when triggered by an OMR prefix
+    // change.
+
+    VerifyOrExit(IsRunning());
+    VerifyOrExit(mHostInfo.IsAutoAddressEnabled());
+
+    VerifyOrExit(aAddress.GetOrigin() == Ip6::Netif::kOriginSlaac);
```

Comment:
Just to check: so in case a 2nd SLAAC address gets added, and the 1st SLAAC address remains unchanged, there will be  a SRP Update reregistration that adds the 2 SLAAC addresses in AAAA records, correct?

Reply:
Yes. Exactly.